tests/experimental/nn/test_sync_batchnorm.py

```diff
@@ -36,40 +36,48 @@ def wrapper(*args, **kwargs):
 
 
 def check_parity(torch_bn, fs_bn, x):
-    yh = torch.ones_like(x)
-    torch_y = torch_bn(x)
-    fs_y = fs_bn(x)
+    yh = torch.randn_like(x)
+    torch_x = x.detach()
+    torch_x.requires_grad = True
+    torch_y = torch_bn(torch_x)
     torch_y.backward(yh)
+    fs_x = x.detach()
+    fs_x.requires_grad = True
+    fs_y = fs_bn(fs_x)
     fs_y.backward(yh)
     assert torch.allclose(torch_y, fs_y), f"{torch_y} != {fs_y}"
     assert torch.allclose(torch_bn.running_mean, fs_bn.running_mean), f"{torch_bn.running_mean} != {fs_bn.running_mean}"
     assert torch.allclose(torch_bn.running_var, fs_bn.running_var), f"{torch_bn.running_var} != {fs_bn.running_var}"
-    assert torch.allclose(torch_bn.weight, fs_bn.weight), f"{torch_bn.weight.grad} != {fs_bn.weight.grad}"
-    assert torch.allclose(torch_bn.bias, fs_bn.bias), f"{torch_bn.bias.grad} != {fs_bn.bias.grad}"
-    # TODO(msb) currently disabled due to PyTorch bug: https://github.com/pytorch/pytorch/issues/57796
-    # assert torch.allclose(torch_bn.weight.grad, fs_bn.weight.grad), f"{torch_bn.weight.grad} != {fs_bn.weight.grad}"
+    assert torch.allclose(torch_bn.weight, fs_bn.weight), f"{torch_bn.weight} != {fs_bn.weight}"
```

Comment:
btw, torch.testing.assert_allclose() is a much better method on asserting tensors.

Reply:
Wow! torch.testing.assert_allclose() is amazing. Thanks for introducing me to it!